decentraland/screens/BuyLandScreen.js

```diff
@@ -53,16 +60,29 @@ export class BuyLandScreen extends React.Component {
 
     const onSuccess = () => {
       showInfo(`${typeDescription} bought successfully.`);
-      removeLandForSale(landForSale);
-      addToMyAssetsList(asset);
+    };
+
+    const onError = () => {
```

Comment:
In a future version I wonder whether we want the error payload to contain all the data we'd need to use to react to the error event (here, the asset in question)

Reply:
Could you clarify your point better?

Reply:
Oh, like if this were

`onError = (asset OR landForSale) => {`

and then below we used that data that was passed into the error handler to call actions rather than variables from a more global scope.

Reply:
Done.

Reply:
Cool, thanks! Resolving this.